libraries/Microsoft.Bot.Builder/IBotTelemetryClient.cs

```diff
@@ -64,6 +64,14 @@ public interface IBotTelemetryClient
         /// <param name="properties">Named string values you can use to search and classify events.</param>
         void TrackTrace(string message, Severity severityLevel, IDictionary<string, string> properties);
 
+        /// <summary>
+        /// Logs a dialog entry / as an Application Insights page view.
+        /// </summary>
+        /// <param name="dialogName">The name of the dialog to log the entry / start for.</param>
+        /// <param name="properties">Named string values you can use to search and classify events.</param>
+        /// <param name="metrics">Measurements associated with this event.</param>
+        void TrackDialogView(string dialogName, IDictionary<string, string> properties = null, IDictionary<string, double> metrics = null);
+
```

Comment:
unfortunately, this is a breaking change.  You will need to add a IBotTelemetryClient2 or ITrackDialogView or something.

Reply:
we should switch to a BotTelemetryClient abstract base class

---
In reply to: [384199303](https://github.com/microsoft/botbuilder-dotnet/pull/3440#discussion_r384199303) [](ancestors = 384199303)

Reply:
good catch. Will re-visit this tomorrow.